interfaces/builtin/docker.go

```diff
@@ -0,0 +1,180 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"github.com/snapcore/snapd/interfaces"
+)
+
+var dockerPermanentSlotAppArmor = []byte(`
```

Comment:
Not for `[]byte`, but I can switch these to `string` and use a cast in the functions instead -- would that be preferable?
```
interfaces/builtin/docker.go:96: const initializer ([]byte)("\n# Description: Allow operating as the Docker daemon. Reserved because this\n#  gives privileged access to the system.\n# Usage: reserved\n\n# Allow sockets\n/{,var/}run/docker.sock rw,\n/{,var/}run/docker/     rw,\n/{,var/}run/docker/**   mrwklix,\n/{,var/}run/runc/       rw,\n/{,var/}run/runc/**     mrwklix,\n\n# Wide read access to /proc, but somewhat limited writes for now\n@{PROC}/ r,\n@{PROC}/** r,\n@{PROC}/[0-9]*/attr/exec w,\n@{PROC}/sys/net/** w,\n@{PROC}/[0-9]*/cmdline r,\n@{PROC}/[0-9]*/oom_score_adj w,\n\n# Wide read access to /sys\n/sys/** r,\n# Limit cgroup writes a bit\n/sys/fs/cgroup/*/docker/   rw,\n/sys/fs/cgroup/*/docker/** rw,\n\n# Allow tracing ourself (especially the \"runc\" process we create)\nptrace (trace) peer=@{profile_name},\n\n# Docker needs a lot of caps, but limits them in the app container\ncapability,\n\n# Docker does all kinds of mounts all over the filesystem\n/dev/mapper/control rw,\n/dev/mapper/docker* rw,\n/dev/loop* r,\n/dev/loop[0-9]* w,\nmount,\numount,\n\n# After doing a pivot_root using <graph-dir>/<container-fs>/.pivot_rootNNNNNN,\n# Docker removes the leftover /.pivot_rootNNNNNN directory (which is now\n# relative to \"/\" instead of \"<graph-dir>/<container-fs>\" thanks to pivot_root)\npivot_root,\n/.pivot_root[0-9]*/ rw,\n\n# file descriptors (/proc/NNN/fd/X)\n/[0-9]* rw,\n# file descriptors in the container show up here due to attach_disconnected\n\n# Docker needs to be able to create and load the profile it applies to containers (\"docker-default\")\n# XXX We might be able to get rid of this if we generate and load docker-default ourselves and make docker not do it.\n/sbin/apparmor_parser ixr,\n/etc/apparmor.d/cache/ r,\n/etc/apparmor.d/cache/.features r,\n/etc/apparmor.d/cache/docker* rw,\n/etc/apparmor/parser.conf r,\n/etc/apparmor/subdomain.conf r,\n/sys/kernel/security/apparmor/.replace rw,\n\n# We'll want to adjust this to support --security-opts...\nchange_profile -> docker-default,\nsignal (send) peer=docker-default,\nptrace (read, trace) peer=docker-default,\n\n# Graph (storage) driver bits\n/dev/shm/aufs.xino rw,\n\n#cf bug 1502785\n/ r,\n") is not a constant
interfaces/builtin/docker.go:107: const initializer ([]byte)("\n# Description: Allow using Docker. Reserved because this gives\n#  privileged access to the service/host.\n# Usage: reserved\n\n# Obviously need to be able to talk to the daemon\n/{,var/}run/docker.sock rw,\n\n@{PROC}/sys/net/core/somaxconn r,\n") is not a constant
interfaces/builtin/docker.go:112: const initializer ([]byte)("\n# The Docker daemon needs to be able to launch arbitrary processes within containers (whose syscall needs are unknown beforehand)\n@unrestricted\n") is not a constant
interfaces/builtin/docker.go:117: const initializer ([]byte)("\nsetsockopt\nbind\n") is not a constant
```